[MRG] 0.18.1 backports

amueller opened the pull request:
This is an update of the 0.18.X branch with what I want to backport for 0.18.1.
It includes bugfixes and documentation changes.
It actually includes most changes from master, apart from some minor additions and API changes.

ping @jnothman @ogrisel @GaelVaroquaux.

There will be more in 0.18.X that hasn't been merged to master, see https://github.com/scikit-learn/scikit-learn/pulls?q=is%3Aopen+is%3Apr+milestone%3A0.18.1

Commit:
FIX n_iter_without_progress and min_grad_norm in TSNE

Adds tests for n_iter_without_progress and min_grad_norm

fukatani committed:
Fix _get_importances. (#7487)

Not to use hasattr.

TomDLT committed:
Fix typo in PCA docstring

iamprakashom committed:
[MRG+1] minor link fix for SAG paper. (#7523)

* Link to SAG paper updated

link to SAG paper in `References` section updated (#7512).

* Link to SAG paper updated

dead link of SAG paper in Logistic Regression user guide updated.
resolves #7512

* SAG paper cited in docstring

SAG paper cited in `Reference` section of docstring(#7512).

pieteradejong committed:
[MRG+1] minor edits to multiclass documentation (#5837)

* minor edits to multiclass documentation

* minor text formatting and language modifications

* minor word changes, sentence clarifications

* attempt to fix ci test

* restored previous paragraph

* minor grammatical correction

* minor grammar fixes

jnothman committed:
DOC paraphrase multiclass description

rombiee committed:
[MRG + 1] Raised value error in HashingVectorizer and CountVectorizer (#7507)

* Raised value error in HashingVectorizer and CountVectorizer

* error statements fixed

* added unit tests

* added unit tests and value error in tfidvect

* fixed pep8 errors

* changed assertion type

* removed fit assert from hashingvectorizer

* changed test names

* unit tests changes and redundancies removed

charltonaustin committed:
[MRG+1] Fixing error with step parameter #7467 (#7469)

* Fixing error with step parameter #7467. Basically converting the step parameter the same way RFE does.

* Adding in an explanation to the test case.

* Moving the location of where we set the data.

* Adding in an assertion against the data.

julienaubert committed:
learning_curve takes error_score parameter and propagates it. (#7397)

raghavrv committed:
BUGFIX _store train_scores only if return_train_score is True (#7535)

rubik committed:
DOC fix typo in NearestNeighbors docstring (#7545)

carlsonp committed:
extra word in code comment (#7561)

jorahn committed:
fix typo in RandomizedSearchCV docs (#7564)

add missing ' in RandomizedSearchCV docs

anguswilliams91 committed:
DOC return values of sample() function in mixture/base.py (#7571)

amueller committed:
add numpy versions for backports / fixes

iabhi7 committed:
Rangespan testimonial has misformatted link (#7576)

jasmainak committed:
DOC FIX: update warning about random seed with scipy (#7577)

bthirion committed:
[MRG+2]: Fix Ridge GCV with centered data (#6178)

kathyxchen committed:
[MRG+1] Docs: refer users to the other encoders to do one hot encoding for labels. (#7315)

* refer users to the other encoders to do one hot encoding for labels.

* added to the 'see more' for labelbinarizer, multilabelbinarizer, and labelencoder' as well as an example to multilabel binarizer

* added note about y labels to the OneHotEncoder docstring

* removed example from MultiLabelBinarizer

* documentation should specify LabelBinarizer, not MultiLabelBinarizer in OHE

nelson-liu committed:
[MRG + 1] fix test_20news_vectorized (#7431)

Run test if dataset has already been downloaded rather than always skipping it

Ajackster committed:
[MRG] DOC Fixed the display of search results (#7517) (#7560)

* Fixed search not expanding page

* Changed the comment to warn users why not to leave fixed height on header

jkarno committed:
DOC add spaces before colons in docstrings (#7589)

zhengruifeng committed:
[MRG] Fix GaussianProcess batch predict in Py3k  #7329 (#7330)

* fix batch

* add test for eval_MSE

elliott2081 committed:
changed address for nose, additionally, should nose2 be considered for testing since the nose project web site suggests new projects should consider using Nose2, or py.test?

TomDLT committed:
DOC add link in the contributing doc for rebasing on master (#7541)

elliott2081 committed:
got rid of extra line

gszpak committed:
[MRG] FIX Calling fit_transform instead of transform in Pipeline's fit_predict (#7585)

* BUGFIX Calling fit_transform instead of transform in Pipeline's fit_predict (#7558)

* PEP8 fixes in test_fit_predict_on_pipeline

* Added comment explaining separate estimators for pipeline in test_fit_predict_on_pipeline

jkarno committed:
[MRG+2] DOC add info to conventions for multi-label fitting (#7519)

* DOC add info to conventions for multi-label fitting

* DOC move to multilabel section and small edits

* DOC clean multilabel examples and clean information

* DOC fix line lengths for multiclass

souravsingh committed:
Update install instructions for Windows (#7550)

jakevdp committed:
BUG: handle integer inputs in LLE (fixes #6033) (#6282)

rushter committed:
Fix test suites (#6348)

nelson-liu committed:
fix an indentation issue in whatsnew (#7605)

NelleV committed:
DOC numpydoc requires space between name and column (#7601)

This patch fixes a tiny mistake in the contirbuting guidelines of the See also
section. The space between the function name and description was missing.

chenhe95 committed:
[MRG + 1] Added some documentation for loading external datasets (Issue 3808) (#7516)

* Update tutorial.rst

* Update tutorial.rst

* Update tutorial.rst

* Update tutorial.rst

* Update index.rst

* Update index.rst

* Update tutorial.rst

* Update tutorial.rst

* Update tutorial.rst

* Update faq.rst

* Update faq.rst

* Divided in two cases (standard columnar and misc data)

I also added a preprocessing note at the end

* Update tutorial.rst

* Update faq.rst

* Update index.rst

Also added some references that were in the original FAQ and pointed the FAQ to here

* Update index.rst

Added the information from the removed part of the FAQ because I felt that the FAQ version was better than the explanation I gave.

* Update index.rst

reference to skimage and also has sklearn.preprocessing.OneHotEncoder instead of OneHotEncoder

* Update index.rst

* Update index.rst

Changed with @jnothman's feedback 
https://github.com/scikit-learn/scikit-learn/pull/7516/files/d16ac523ed404188fc1f2529ac89050d4a974e3f

* Update faq.rst

* optimized file formats added to datasets/index.rst

Note: if you manage your own numerical data it is recommended to use an optimized file format such as HDF5 to reduce data load times. Various libraries such as H5Py, PyTables and pandas provides a Python interface for reading and writing data in that format.

- From the FAQ

* faq.rst: Moved the comment in bunch section to datasets index

This comment has been moved to the datasets index in the external_datasets section:
Note: if you manage your own numerical data it is recommended to use an optimized file format such as HDF5 to reduce data load times. Various libraries such as H5Py, PyTables and pandas provides a Python interface for reading and writing data in that format.

* Update index.rst

Included all changes mentioned by @amueller and @jnothman

* Update faq.rst

* Update faq.rst

lynchkl committed:
[MRG + 2] updated iris svm example colormap so the three species can be distinguished (#7611)

* updated iris svm colormap so the three species can be distinguished

* updated colormap to viridis, which is robust to colorblindness

* changing colormap to coolwarm

kgilliam125 committed:
[MRG] Documentation describes the Need Contributor label (#7599)

amueller committed:
DOC pos_label can be string (#7607)

ivallesp committed:
TST ensure relationship between predict_proba and predict_log_proba (#7612)

nelson-liu committed:
[MRG] FIX min_samples_split and min_samples_leaf validation for float vs int (#7604)

lesteve committed:
COSMIT fix typo in docstring

Commit:
DOC fixed avoidable warning of test-doc (#7524)

ogrisel committed:
DOC fix link format in reference for BayesianMixtureModel

jakevdp committed:
Add whats new entry for #6282 (#7629)

jakevdp committed:
Add whats new entry for #6282 (#7629)

sshleifer committed:
[MRG] FIX regression: Pipeline.fit_predict uses fit_params as in 0.17.x (#7621)

Commit:
dev-7360: Explain the meaning of prefix [MRG + 1] and [MRG + 2] in the title of the pull request. (#7636)

* File updated: doc/developers/contributing.rst

toastedcornflakes committed:
Improve robustness of LFW dataset downloader (#6112)

ogrisel committed:
DOC better docstring for TruncatedSVD (#7651)

ogrisel committed:
MAINT make appveyor fail on old builds when PR is update (#6365)

amueller committed:
DOC Removing deprecated DPGMM that was also not rendering correctly from classes. (#7606)

nvavrova committed:
DOC Fixed missing "Next" button (#7641)

GaelVaroquaux committed:
DOC: fix the copybutton on the code blocks (#7634)

The copybutton is the ">>>" on the side of code blocks that enables to
remove the ">>>" at the begining of lines as well as the outputs.

It was broken in the docs of 0.18 due to changes in sphinx. I am
backporting the js used by the Python core docs.

In addition, I am backporting CSS to highlight this button
on hover.

Commit:
dev-7369: Make common metric tests look nicer (#7620)

* Change the tests to yield after performing the _named_check.

* Tests affected:

  -test_averaging_multiclass
  -test_averaging_multilabel
  -test_averaging_multilabel_all_zeroes
  -test_averaging_multilabel_all_ones
  -test_sample_weight_invariance

* File updated: sklearn/metrics/tests/test_common.py

nelson-liu committed:
[MRG] edit releases link from sf to pypi (#7637)

* edit releases link from sf to github

* edit link to pypi

Commit:
[MRG+1] Throw an error with explicit message if n_estimators is not an integer. (#7457)

* Throw an error with explicit message if n_estimators is not an integer.

* Testing for explicit message if n_estimators is not an integer.

* Fixed typo in test for explicit message if n_estimators is an integer.

* Added tests for np.int32 and float input.

* pep8 compliance

* fix function name

* Import numpy to test n_estimators suplied as numpy int32.

amueller committed:
[MRG+3] Fix min_weight_fraction_leaf to work when sample_weights are not provided (#7301)

* fix min_weight_fraction_leaf when sample_weights is None

* fix flake8 error

* remove added newline and unnecessary assignment

* remove max bc it's implemented in cython and add interaction test

* edit weight calculation formula and add test to check equality

* remove test that sees if two parameter build the same tree

* reword min_weight_fraction_leaf docstring

* clarify uniform weight in forest docstrings

* update docstrings for all classes

* add what's new entry

* move whatsnew entry to bug fixes and explain previous behavior

# Conflicts:
#	doc/whats_new.rst

amueller committed:
DOC: trivial rst fix to whats_new.rst (#7511)

Fix for proper rendering

amueller committed:
DOC add what's new for #6178

# Conflicts:
#	doc/whats_new.rst

amueller committed:
[MRG+1] FIX Make sure GridSearchCV and RandomizedSearchCV are pickle-able (#7594)

* FIX Subclass a new MaskedArray which allows pickling even when dype=object

* TST unpickling too

* FIX Use MaskedArray from utils.fixes rather than from numpy

* FIX imports

* Don't assign a variable

* FIX np --> numpy

* Use tostring instead of tobytes for old numpy

* COSMIT pickle-able --> picklable

* use #noqa comment to turn off flake8

* TST/ENH Check if the pickled est's predict matches with the original one's

# Conflicts:
#	sklearn/utils/tests/test_fixes.py

amueller committed:
[MRG + 1] FIX: remove checks from PyFunc distance metric (fixes #6287) (#6288)


# Conflicts:
#	sklearn/neighbors/tests/test_dist_metrics.py

amueller committed:
DOC What's new for #6288

# Conflicts:
#	doc/whats_new.rst

amueller committed:
DOC what's new for #6497 and 0.18.1 section

# Conflicts:
#	doc/whats_new.rst

amueller committed:
[MRG + 1] [TST] (half-cosmetic) use less nose.tools import to simplify future transition to py.test (#7384)

* use less nose.tools import to simplify future transition to activly developing test suites/runners

* assert_equal -> assert_array_equal in test_feature_hasher_pairs_with_string_values

and one missed ImportError that should be replaced with AttributeError

* test for py2.6 compat with except AttributeError

* fix importing of SkipTest

* force using nose in python2.6 for now

* there was no assert_dict_equal in py2.6. but we can use assert_equal

although failed test will look a little bit ugly

* remove nose imports from doc/datasets

# Conflicts:
#	sklearn/neighbors/tests/test_dist_metrics.py
#	sklearn/utils/tests/test_fixes.py

amueller commented:
@jnothman I know we discussed the 0.18.1 timeline somewhere. I think it's doable within the next two weeks. The PRs are mostly there and there are very few issues without PRs.

jnothman commented:
That's great, but my availability over the next two weeks is very limited,
so don't depend on me :)

On 15 October 2016 at 08:25, Andreas Mueller <notifications@github.com>
wrote:

> @jnothman <https://github.com/jnothman> I know we discussed the 0.18.1
> timeline somewhere. I think it's doable within the next two weeks. The PRs
> are mostly there and there are very few issues without PRs.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/scikit-learn/scikit-learn/pull/7672#issuecomment-253923429>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAEz69UtgN8JKcx-KgsH7Tmf8fMQb4WMks5qz_NTgaJpZM4KXccV>
> .
>

andrewcsmith committed:
DOC: fix return value doc for mixture.base._e_step (#7682)

amueller commented:
@jnothman noted, no worries :)

Commit:
warning for PCA with sparse input (#7649)

jstriebel committed:
[MRG+1] CircleCI timeout extended (#7693)

* Circle CI: adding 1h timeout for build_doc.sh

* Circle CI: typo + trigger build

* CircleCI: trigger build again

FERRIA committed:
DOC Changed Contributor's Guide to Development Guide #7690 (#7691)

ljwolf committed:
switch to multinomial composition for mixture sampling

lesteve committed:
DOC mention JOBLIB_START_METHOD in crash/freeze FAQ

jnothman committed:
DOC use combined unicode chars in author names (#7706)

ljwolf committed:
add shape assertions to test

olologin committed:
FIX #6420: Cloning decision tree estimators breaks criterion objects (#7680)

lesteve committed:
Update joblib to 0.10.3 version (#7696)

amueller committed:
add missing links for users to whatsnew

JungeAlexander committed:
[MRG + 1] Move n_iter and get_params invariance tests to common estimator_checks (#7677)

* Test get_params invariance in common estimator tests

Remove test_get_params_invariance() from `test_common.py` and add
test call to _yield_all_tests() in `estimator_checks.py` to make
sure that get_params(deep=False) of a given Estimator returns a
subset of get_params(deep=True).

Compared to test_get_params_invariance(), it is NOT tested anymore
whether the given Estimator has an attribute get_params since
class BaseEstimator in `base.py` defines such an attribute
for each Estimator.

Partially addresses issue #7533
Also related to issue #4465

* Move test_transformer_n_iter() to estimator_checks.py

Remove the test test_transformer_n_iter() from tests/test_common.py
and perform the test logic in utils/estimator_checks.py instead.
Specifically, the method _yield_transformer_checks() now yields
check_transformer_n_iter() as part of the set of tests for
transformers.

test_transformer_n_iter() tests that that transformers with an
attribute max_iter, return the attribute of n_iter at least 1.

Partially addresses latter part of issue #7533

* Move test_non_transformer_estimators_n_iter() to estimator_checks.py

Remove the test_non_transformer_estimators_n_iter() from
tests/test_common.py; perform the test logic in
utils/estimator_checks.py instead.
Specifically, the method _yield_non_meta_checks() now yields
check_non_transformer_estimators_n_iter().

test_transformer_n_iter() tests that that estimators that are not
transformers with an attribute max_iter, return the attribute n_iter
of at least 1.

NOTE: The current implementation makes said test run for more
estimators than before this commit.
For some of these estimators, the test fails. This needs to be addressed
(see FIXME in line 111-115 of utils/estimator_checks.py for a potential
place to start).

Partially addresses latter part of issue #7533

* Fix check_non_transformer_estimators_n_iter calls

test_transformer_n_iter() test is now only run for
estimators where the test is applicable.

Partially addresses latter part of issue #7533

* Run check_non_transformer_estimators_n_iter on multi-class estimators

To do this, use helper method multioutput_estimator_convert_y_2d.
Also remove multi_output parameter from
check_non_transformer_estimators_n_iter since this parameter is not
used anywhere and corresponding cases should be handled by said
helper method.

Also, some pep8 line length fixes.

* Fix documentation for n_iter tests

There was some confusion between attributes and parameters.
Also rename n_iter to n_iter_

lesteve committed:
Use n_components=3 to test actual regression

n_components and n_features were equal and one was used for the other in
some places.

affanv14 committed:
[MRG+2] adding multilabel support for score_func (#7676)

* added multilabel support for score function

* added test for multilabel score function

* updated whats_new.rst

* updated whats_new.rst with working link

kgilliam125 committed:
[MRG] Added warning on keyboard interrupt during MLP fit (#7614)

amueller committed:
removed parameter that was documented as attribute (#7711)

mblondel committed:
Fix typo in OMP author name.

First author is Stéphane Mallat.

tguillemot committed:
[MRG+1] Reorder EllipticEnvelope docstring. (#7734)

* Reorder EllipticEnvelope docstring.

* Fix pep8

srivatsan-ramesh committed:
[MRG + 1] Printing the total time in cross_validation (#7640)

* print score+fit time instead of just score time when doing cross_validation

* reducing line size

* More clearer log message

chkoar committed:
Address #7733 - MultiTaskElasticNet user guide links to MultiTaskLasso (#7741)

amueller committed:
FIX #6420: Cloning decision tree estimators breaks criterion objects (#7680)


# Conflicts:
#	doc/whats_new.rst

amueller committed:
[MGR + 2] fix selectFdr bug (#7490)


# Conflicts:
#	doc/whats_new.rst

bburns committed:
r2_score - add more doctest examples (#7727)

Add simple examples showing perfect score, expected value score, and negative score.

kgilliam125 committed:
[MRG + 1] Clarified error msg in plot_partial_dependence (#7673)

* Clarified error msg in plot_partial_dependence

* Changed err msg for feature[i] out of range. Updated docs.

* Error message shows invalid value.

GaelVaroquaux committed:
BF: avoid importing from inside joblib (#7731)

Debian replaces externals.joblib by an import to joblib, because they
hate duplication. Hence importing from inside joblib doesn't work, and
the line that I removed greats a bug.

See
http://lists.alioth.debian.org/pipermail/neurodebian-users/2016-October/001093.html

erikcs committed:
[MRG + 1] ElasticNetCV: raise ValueError if l1_ratio=0 (#7591)

Raise ValueError if l1_ratio=0 in ElasticNetCV and alphas=None

Deborah-Digges committed:
[MRG+2] logit -> logistic in plot_logistic.py and minor visual improvements to the plot. (#7730)

* Small correction of logit->logistic

The function `1 / (1 + np.exp(-x))` is the logistic function. 
The logit function is the inverse of the logistic function : `log(x/(1-x))`

* Add axes ticks, legend and colors to the plot

* Fix flake8 errors

* Rename legend labels as per review

* Fix pep8 error

GaelVaroquaux commented:
Travis is failing in an ugly way. Is this a cythonization problem?

JPFrancoia committed:
[MRG] Correcting length of explained_variance_ratio_, eigen solver (#7632)

jnothman committed:
[MRG] DOC :issue: role to simplify what's news (#7657)

raghavrv commented:
>  Is this a cythonization problem?

It seems like it...

kmike committed:
DOC use target_names over named categories in 20newsgroups example (#7423)

raghavrv committed:
DOC Correct linking of TruncatedSVD (#7749)

raghavrv commented on sklearn/decomposition/pca.py:
```diff
@@ -115,6 +116,9 @@ class PCA(_BasePCA):
     truncated SVD.
 
+    Notice that this class does not support sparse input. See
+    :ref:`<TruncatedSVD>` for an alternative with sparse data.
```
Done in #7749...

polmauri committed:
[MRG + 1] FIX raise an error message when n_groups > number of groups (#7681) (#7683)

* FIX raise an error message when n_groups > actual number of groups (#7681)

This change addresses issue #7681:
- Raise ValueError when n_groups > actual number of unique groups in LeaveOneGroupOut and LeavePGroupsOut.
- Add unit test.

* Make requested changes

- Check error message with `assert_raise_message`
- Pass parameters to `assert_raise_message` instead of defining functions

* Update condition and exception message

amueller committed:
[MRG+1] replaced some assert_true(np.allclose(x, y)) with assert_almost_equal (#7742)

asanakoy committed:
[MRG + 1] fix bug with negative values in cosine_distances (#7732)

* fix bug with negative values in cosine_distances
clip distances to [0, 2]
set distances between vectors and themselves to 0

* add test

* add test on big random matrix

* use np.diag_indices_from instead of slicing

perimosocordiae committed:
[MRG+1] BUG: MultiLabelBinarizer.fit_transform sometimes returns an invalid CSR matrix (#7750)

* BUG: MultiLabelBinarizer makes invalid CSR matrix

See https://github.com/scipy/scipy/issues/6719 for context.

The gist is that the `inverse` array may have a different dtype than `yt.indices`, which causes trouble down the line because, in those cases, `yt.indices` and `yt.indptr` have different dtypes.

Alternately, we could insert `yt.check_format(full_check=False)` after modifying the sparse matrix members.

* Fixing for old numpy

Older versions don't support kwargs for `astype`

* Adding tests

* line-wrapping

* adding comment to tests

[ci skip]

* added rationale comment

[ci skip]

GaelVaroquaux commented:
There are test failures that concern the LocalOutlierFactor. I am quite pluzlled, as would think that LOF was not in the 0.18.1 branch. Is this a backport error, or am I confused?

jnothman commented:
Is there a build tool where we inadvertently merge master?

jnothman commented:
It's not in the backport diff, so I'm assuming it's a CI error, but haven't found its source.

amueller commented:
LOF shouldn't be here.. hum

amueller commented:
It does ``git fetch origin +refs/pull/7672/merge`` thought the merge-base should be ``0.18.X``, right?

amueller commented:
Hm the checkout gives the right results, it might be a caching issue?

amueller commented:
appveyor and circleci fail on cythonize.

amueller commented:
There's an additional ``os.remove(HASH_FILE)`` in this branch and I have no idea where it came from...

raghavrv commented:
That was from [my patch](https://gitter.im/scikit-learn/scikit-learn?at=580f8906df0fb2245fc1950a) to clear the travis cache. It removes the `cythonize.dat` file to force recythonization... It is added by the `NOMERGE recythonize all` commit which is expected to be deleted at the end. I'm not sure if that is what messes up the CIs? (As it affects only the travis cache)

raghavrv commented:
Ah it tries to remove it when it doesn't exist. If you delete that commit, the CIs should be good I think...

amueller commented:
I'm just confused by github not showing me any history on the cythonize file in master: https://github.com/scikit-learn/scikit-learn/blob/master/build_tools/cythonize.py

amueller commented:
ah it was the commit you suggested to kill the cache... right.

Erotemic committed:
[MRG+2] Fixed n**2 memory blowup in _labels_inertia_precompute_dense (#7721)

jkarno committed:
[MRG] ENH faster sample_without_replacement for recent numpy (#7703)

waterponey committed:
[MRG] CI build pdf docs when master or version branch (#7747)

kmike committed:
DOC parameter is called "sample_weight", not "sample_weights" (#7782)

waterponey committed:
[MRG+2] DOC framework for keeping API refs for deprecated classes/funcs (#7725)

* DOC framework for keeping API refs for deprecated classes/funcs

* DOC tagging deprecated for 0.20

* suggestion for LDA/QDA deprecation

* simplify deprecation message for GaussianProcess

* simplify deprecation messages

* fixup avoid import QuadraticDiscriminantAnalysis in qda.QDA (and similar for LDA).

* fixup test alias lda.LDA is instance of LinearDiscriminantAnalysis

raghavrv committed:
[MRG + 2] FIX Be robust to non re-entrant/ non deterministic cv.split calls (#7660)

iKevinY committed:
DOC Minor changes to README (#7791)

kaushlakers committed:
[MRG] DOC Isolation Forest decision function doc - Clarify that it computes normality not abnormality. (#7792)

ditenberg committed:
Change fit() docstrings (#7783)

In reference to #7772

Commit:
[MRG + 1] rfecv: verbosity: Set verbose threshold low bound from 2 to 1 (#7644)

* rfecv: verbosity: Set verbose threshold low bound from 2 to 1

* rfecv: verbosity: Add test checking verbose=1 produce output

* rfecv: verbosity: Change docstring to comment

* rfecv: verbosity: Fix test, missing seek

waterponey committed:
fix rst errors for pypi (#7800)

Commit:
DOC remove duplicate versionadded

Commit:
COSMIT delete untrue and obsolete comment

lesteve committed:
[MRG+2] BUILD Rewrite setup.py files to handle cython dependencies (#7719)

* Rewriting of cythonization in setup.py

By using Cython.Build.cythonize and switching between .c and .pyx files
as appropriate cython dependencies are correctly taken into account.

* Use cythonize once on the root config rather than in each subpackage

* Fix for Windows

* Remove caching from Travis

Cython dependencies are taken care of by Cython.Build.cythonize and
based on file timestamps, so .C and .so files will always be rebuild
from scratch on each build in Travis.

* Specify .pyx in setup.files for cython generated extensions

More natural this way. Tweak the extensions to generate from .c and .cpp
files for a release.

* COSMIT Remove commented out code

* Check cython version is greater than 0.23

* COSMIT better names for functions

* flake8 fix (imported module not at top of file)

* Install cython 0.23 for Python 2.6

now that cython >= 0.23 requirement is enforced in setup.py

* Use module constant for minimum required cython version

* Fix Travis install.sh

No easy way to put comments inside multi-line command

lesteve committed:
Fix typo in testimonials.rst

jnothman committed:
DOC More uses of :issue: in whats_new

maniteja123 committed:
[MRG] MAINT: Return self for fit in Spectral Biclustering and CoClustering (#7814)

raghavrv committed:
[MRG] FIX Validate and convert X, y and groups to ndarray before splitting (#7593)

kiote committed:
[MRG] TST Ensure __dict__ is unmodified by predict, transform, etc (#7553)

* Add test for __dict__ for estimator checks

  check that "predict", "transform", "decision_function" or
"predict_proba" methods do not change the state of __dict__ of any
estimator

  see #7297

* Add a test to test_check_estimator

  that shows that check_estimator fails on an estimator that violates this

* Fixed bug where NMF's n_iter_ attribute was set by calls to transform

srivatsan-ramesh committed:
[MRG+2] Fix for OvR partial_fit bug (#7786)

* mini-batch can now contain less number of classes than actual data

* added tests where mini batches doesn't contain all classes

srivatsan-ramesh committed:
DOC Added bug fix #7786 to whats_new.rst (#7830)

btdai committed:
[MRG + 2] FIX bug in svm's decision values when `decision_function_shape` is `'ovr'` (#7724)

lesteve committed:
Disable git pager in flake8_diff.sh

lesteve commented:
I think we should move the warning about dropping Python 2.6 support to the 0.18.1 section to avoid confusion:
```
.. topic:: Last release with Python 2.6 support                                                                                       
                                                                                                                                      
    Scikit-learn 0.18 will be the last version of scikit-learn to support Python 2.6.                                                 
    Later versions of scikit-learn will require Python 2.7 or above.   
```

tguillemot committed:
[MRG + 1] Update funding and website. (#7836)

* Fix funding and website.

* Update images and fix typos.

* Fix images problem.

guoci committed:
DOC markup typo in regression.py (#7837)

agramfort committed:
DOC : fix docstring of AIC/BIC in GMM

srivatsan-ramesh committed:
[MRG+1] label binarizer not used consistently in CalibratedClassifierCV (#7799)

* label binarizer not used consistently in CalibratedClassifierCV

* changed position of classes argument to make old tests run

* moved parameter to constructor and added test

* added test where train set doesnt have all classes

* CalibratedClassifierCV can now handle cases where train set doesnt contain all labels

* fixing flake error

* fixing line lengths

* removing np.full()

* from __future__ import division for py2.7

* change is test file

* added an extra test and removed a test with Ridge

* stronger test

* whats new entry

dalmia committed:
[MRG + 1] DOC adding :user: role to whats_new (#7818)

* DOC adding :user: role to whats_new

* DOC correcting the improper username

* DOC adding exceptions for ImportError in sphinx_issues.py

* DOC added import for split_explicit_title

* DOC reverting to the old format for core devs

* DOC reverting changes for more users

* DOC removing extra lines and blank lines

* DOC Adding missing link target for user

* DOC fixing more links

* DOC adding missing links for new contributions

* DOC adding space before brackets in the user links

* DOC reverting a wrong change

* DOC adding final changes

dkirkby committed:
Fix docstring typo (#7844) n_features --> n_components

dalmia committed:
[MRG+2] DOC adding separate `fit()` methods (and docstrings) for DecisionTreeClassifier and DecisionTreeRegressor (#7824)

* DOC adding separate fit() functions

* DOC adding keywords to arguments of super()

* DOC removing trailing whitespaces

* DOC specifying the type of class labels

* DOC removing docstring from BaseDecisionTree.fit

amueller committed:
[MRG + 2] Fixed parameter setting in SelectFromModel (#7764)

* Fixed cloning ``estimator`` again when calling fit a second time in SelectFromModel

* fix link in whatsnew

raghavrv committed:
[MRG + 2] FIX LogisticRegressionCV to correctly handle string labels (#5874)

* TST if LogisticRegressionCV handles string labels properly
* TST Add a test with class_weight dict
* ENH Encode y and class_weight dict
* Better variable names
* TYPO casses --> classes
* FIX Use dict comprehension; classes_labels --> classes
* Revert dict comprehension (for Python 2.6 compat)
* MNT reorder validation to improve clarity
* Add whatsnew entry

amueller committed:
fixed whatsnew cherry-pick mess (somewhat)

amueller committed:
[MGR + 2] fix selectFdr bug (#7490)


# Conflicts:
#	doc/whats_new.rst

amueller committed:
sync whatsnew with master

amueller commented:
@jnothman @agramfort @GaelVaroquaux @ogrisel this is the final state for 0.18.1
@jnothman anything I'm missing? Sorry I've been offline for a couple of days.

amueller commented:
looking at tests now

amueller committed:
fix learning_curve test that I messed up in cherry-picking the "reentrant cv" PR.

amueller committed:
add the python2.6 warning to 0.18.1

amueller committed:
fix merge conflict mess in whatsnew

amueller commented:
I'm not gonna change the pep8 in the setup for this, I think. If it runs through now I'll release tomorrow.

amueller commented:
merged manually as I didn't wanna squash
